tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -1076,6 +1092,43 @@ export function reducers(state: MetricsState | undefined, action: Action) {
   return composeReducers(reducer, namespaceContextedReducer)(state, action);
 }
 
+function getMinMaxTimeSelection(timeSeriesData: TimeSeriesData): TimeSelection {
```

Comment:
The name makes me think it's the min/max _of_ time selection. But it's "_of_ the experiment _for_ time selection)
Propose something like `getMinMaxStepForTimeSelection` or just `getStepMinMax` ?

Another more high level question, what is  the output different compared to the ngrx state? 

Updated here
[tensorboard/webapp/metrics/store/metrics_reducers.ts](https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/webapp/metrics/store/metrics_reducers.ts;l=790?q=tensorboard%2Fwebapp%2Fmetrics%2Fstore%2Fmetrics_reducers.ts&ss=tensorflow%2Ftensorboard)

Note: I haven't check the implementation details yet but the output looks like what you want

Reply:
> The name makes me think it's the min/max of time selection. But it's "of the experiment for time selection)
Propose something like getMinMaxStepForTimeSelection or just getStepMinMax ?

`getStepMinMax` makes sense to me.
Looking at `getMinMaxTimeSelection` I would hope it was read as
"action" -> "get"
"noun" -> "Min Max TimeSelection"
i.e. get a time selection containing the min and max step.

> Another more high level question, what is the output different compared to the ngrx state?

This is a pure function, there is no relationship to the ngrx state.